packages/@aws-cdk/aws-wafv2/lib/ipset.ts

```diff
@@ -0,0 +1,111 @@
+import * as cdk from '@aws-cdk/core';
+import * as waf from './index';
+
+export enum IPAddressVersion {
+    IPV4 = 'IPV4',
+    IPV6 = 'IPV6'
+}
+
+export enum IPSetScope {
+    CLOUDFRONT = 'CLOUDFRONT',
+    REGIONAL = 'REGIONAL'
+}
+
+export interface IIPSet {
+    /**
+     * A friendly name of the IP set. You cannot change the name of an IPSet
+     * after you create it.
+     */
+    readonly ipSetName: string;
+}
+
+export abstract class IPSetBase extends cdk.Resource implements IIPSet {
+    /**
+     * A friendly name of the IP set. You cannot change the name of an IPSet
+     * after you create it.
+     */
+    public abstract readonly ipSetName: string;
+}
+
+export interface IPSetProps {
```

Comment:
This might lead to trouble in jsii. It might not translate this to a struct. @RomainMuller? Do we have an annotation to force an interface whose name starts with `I` to a struct?

Reply:
Seems this was updated.